File: ios/MazadatPicker.m
//
//  MazadatPicker.m
//  MazadatPicker
//
//  Created by Karim Saad on 15/11/2022.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(MazadatPicker,NSObject)
RCT_EXTERN_METHOD(increment)

RCT_EXTERN_METHOD(openCamera:(NSString *)title_ aspect_ratio_x:(NSInteger)aspect_ratio_x aspect_ratio_y:(NSInteger)aspect_ratio_y promise:(RCTPromiseResolveBlock)promise reject:(RCTPromiseRejectBlock)reject)
 RCT_EXTERN_METHOD(openGallery:(NSString *)title_ aspect_ratio_x:(NSInteger)aspect_ratio_x aspect_ratio_y:(NSInteger)aspect_ratio_y promise:(RCTPromiseResolveBlock)promise reject:(RCTPromiseRejectBlock)reject)
 RCT_EXTERN_METHOD(editImage:(NSString *)path_ title_:(NSString *)title_ aspect_ratio_x:(NSInteger)aspect_ratio_x aspect_ratio_y:(NSInteger)aspect_ratio_y promise:(RCTPromiseResolveBlock)promise reject:(RCTPromiseRejectBlock)reject)
+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/MazadatPicker.swift
//
//  MazadatPicker.swift
//  MazadatPicker
//
//  Created by Karim Saad on 15/11/2022.
//

import Foundation

@objc(MazadatPicker)
class MazadatPicker: NSObject{
  private var count=0
  @objc
  func increment(){
    count+=1;
    //resolve(count)
  }
  
   @objc
  func openCamera(_ title_:String,aspect_ratio_x:Int,aspect_ratio_y:Int,promise:@escaping RCTPromiseResolveBlock,reject:@escaping RCTPromiseRejectBlock){
    DispatchQueue.main.async {
      let cameraContoller = CameraController(nibName: "CameraController", bundle: nil)
      cameraContoller.modalPresentationStyle = .fullScreen
      cameraContoller.setData(title_: title_, aspectRatioX_: Float(aspect_ratio_x), aspectRatioY_: Float(aspect_ratio_y))
      cameraContoller.setPromise(promise: promise)
      UIApplication.shared.windows.filter {$0.isKeyWindow}.first?.rootViewController?.present(cameraContoller, animated: true, completion: nil)
    }
        
         print("open camera");
  }
  @objc
  func openGallery(_ title_:String,aspect_ratio_x:Int,aspect_ratio_y:Int,promise:@escaping RCTPromiseResolveBlock,reject:@escaping RCTPromiseRejectBlock){
    DispatchQueue.main.async {
      let controller = Gallery(nibName: "Gallery", bundle: nil)
      controller.modalPresentationStyle = .fullScreen
      controller.setData(title_: title_, aspectRatioX_: Float(aspect_ratio_x), aspectRatioY_: Float(aspect_ratio_y))
      controller.setPromise(promise: promise)
      UIApplication.shared.windows.filter {$0.isKeyWindow}.first?.rootViewController?.present(controller, animated: true, completion: nil)
   
    }
     
   
   
    print("open gallery");
  }
      
  @objc
  func editImage(_ path_:String, title_:String,aspect_ratio_x:Int,aspect_ratio_y:Int,promise: @escaping RCTPromiseResolveBlock, reject:@escaping RCTPromiseRejectBlock){
    DispatchQueue.main.async {
      let controller = EditPhoto(nibName: "EditPhoto", bundle: nil)
      controller.modalPresentationStyle = .fullScreen
      controller.setData(title_: title_, aspectRatioX_: Float(aspect_ratio_x), aspectRatioY_: Float(aspect_ratio_x), fileName: URL(string: path_)!, from: "edit")
      controller.setPromise(promise: promise)
      UIApplication.shared.windows.filter {$0.isKeyWindow}.first?.rootViewController?.present(controller, animated: true, completion: nil)
      
    }
    print("open camera");
  }
}
